Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp

```diff
@@ -0,0 +1,356 @@
+
+            // Record old values of the fields to be updated
+            Complex const Ep_old = fields(i,j,k,Ep_m);
+            Complex const Em_old = fields(i,j,k,Em_m);
+            Complex const Ez_old = fields(i,j,k,Ez_m);
+            Complex const Bp_old = fields(i,j,k,Bp_m);
+            Complex const Bm_old = fields(i,j,k,Bm_m);
+            Complex const Bz_old = fields(i,j,k,Bz_m);
+            // Shortcut for the values of J and rho
+            Complex const Jp = fields(i,j,k,Jp_m);
+            Complex const Jm = fields(i,j,k,Jm_m);
+            Complex const Jz = fields(i,j,k,Jz_m);
+            Complex const rho_old = fields(i,j,k,rho_old_m);
+            Complex const rho_new = fields(i,j,k,rho_new_m);
+
+            // k vector values, and coefficients
+            // The k values for each mode are grouped together
+            int const ir = i + nr*mode;
+            amrex::Real const kr = kr_arr[ir];
+            amrex::Real const kz = modified_kz_arr[j];
+
+            constexpr amrex::Real c2 = PhysConst::c*PhysConst::c;
+            constexpr amrex::Real inv_ep0 = 1._rt/PhysConst::ep0;
+            Complex const I = Complex{0._rt,1._rt};
+            amrex::Real const C = C_arr(i,j,k,mode);
+            amrex::Real const S_ck = S_ck_arr(i,j,k,mode);
+            Complex const X1 = X1_arr(i,j,k,mode);
+            Complex const X2 = X2_arr(i,j,k,mode);
+            Complex const X3 = X3_arr(i,j,k,mode);
+            Complex const X4 = X4_arr(i,j,k,mode);
+            Complex const T2 = Theta2_arr(i,j,k,mode);
+
+            // Update E (see WarpX online documentation: theory section)
+            fields(i,j,k,Ep_m) = T2*C*Ep_old
+                        + T2*S_ck*(-c2*I*kr/2._rt*Bz_old + c2*kz*Bp_old)
+                        + X4*Jp + 0.5_rt*kr*(X2*rho_new - T2*X3*rho_old);
+            fields(i,j,k,Em_m) = T2*C*Em_old
+                        + T2*S_ck*(-c2*I*kr/2._rt*Bz_old - c2*kz*Bm_old)
+                        + X4*Jm - 0.5_rt*kr*(X2*rho_new - T2*X3*rho_old);
+            fields(i,j,k,Ez_m) = T2*C*Ez_old
+                        + T2*S_ck*(c2*I*kr*Bp_old + c2*I*kr*Bm_old)
+                        + X4*Jz - I*kz*(X2*rho_new - T2*X3*rho_old);
+            // Update B (see WarpX online documentation: theory section)
+            // Note: here X1 is T2*x1/(ep0*c*c*k_norm*k_norm), where
+            // x1 has the same definition as in the original paper
+            fields(i,j,k,Bp_m) = T2*C*Bp_old
+                        - T2*S_ck*(-I*kr/2._rt*Ez_old + kz*Ep_old)
+                        + X1*(-I*kr/2._rt*Jz + kz*Jp);
+            fields(i,j,k,Bm_m) = T2*C*Bm_old
+                        - T2*S_ck*(-I*kr/2._rt*Ez_old - kz*Em_old)
+                        + X1*(-I*kr/2._rt*Jz - kz*Jm);
+            fields(i,j,k,Bz_m) = T2*C*Bz_old
+                        - T2*S_ck*I*(kr*Ep_old + kr*Em_old)
+                        + X1*I*(kr*Jp + kr*Jm);
+        });
+    }
+};
```

Comment:
warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```